src/Sentry/Platforms/Cocoa/RuntimeMarshalManagedExceptionIntegration.cs

```diff
@@ -0,0 +1,45 @@
+using ObjCRuntime;
+using Sentry.Extensibility;
+using Sentry.Integrations;
+
+namespace Sentry.Cocoa;
+
+/// <summary>
+/// When AOT Compiling iOS applications, the AppDomain UnhandledExceptionHandler doesn't fire. So instead we intercept
```

Comment:
oh we've been missing out on a category of errors?

Reply:
Essentially, yes. Matt created this workaround to the problem:
https://github.com/getsentry/sentry-dotnet/blob/c20d16245c7e2d2f9056a49780593a391e59eda1/src/Sentry/Platforms/Cocoa/SentrySdk.cs#L13-L17

However that marshalling mode isn't supported when AOT compiling iOS apps. It [causes xamarin to throw an assertion error](https://github.com/xamarin/xamarin-macios/blob/907081f787315704a01c940cf28b46b47db23df0/runtime/runtime.m#L2452-L2462), which is unrelated to the error from the user's code.